### PR TITLE
Bug 1634375 - Merge new upload mechanism into master

### PR DESCRIPTION
**Pull request by badboy**
It's about time!
After I just cut a release it's finally time to bring this back into master.

It's been extensively tested and has been through QA on both Android (inside Fenix) and iOS (thank @travis79).

It's individual pieces have been written by @brizental and followups by me.
It's been reviewed by me and mostly @Dexterp37 lately.

This merge is probably too much to review as one PR, but again the individual pieces all landed one-by-one as smaller PRs on the m16/ping-uploading branch.

🚀 Glean's uploading now! 🚀

### CODE REVIEW

**Comment by badboy** [comment]
spellcheck task and TaskCluster failing because it's missing the updated configs in this branch.
master has the config and it should be green on merge into it. I will still run the iOS tests here once again.

File: docs/mermaid-init.js [hunk 1]
```diff
@@ -0,0 +1 @@
+mermaid.initialize({startOnLoad:true});
```
Comment (Dexterp37) [inline docs/mermaid-init.js:1]: Missing license header?

File: docs/mermaid.css [hunk 2]
```diff
@@ -0,0 +1,351 @@
+/* Flowchart variables */
```
Comment (Dexterp37) [inline docs/mermaid.css:1]: Does this need a license header?

**Comment by badboy** [comment]
Ok, merged back master first to resolve the file conflict. Should now apply cleanly.

**Pull request merged**